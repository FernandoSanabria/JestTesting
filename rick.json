{
    "id":1,"name":"Ricok Sanchez","status":"Alive","species":"Human","type":"","gender":"Male"
}