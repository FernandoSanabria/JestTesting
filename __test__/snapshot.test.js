import {getCharacter} from '../snapshot';
import rick from '../rick.json'

describe('Snapshot', () => {
    test('rick object snapshot test', () => {
        expect(getCharacter(rick)).toMatchSnapshot();
    });

    test('faill snapthot test', () => {
        const user =  {
            id: Math.floor(Math.random()*30)
        }
        expect(user).toMatchSnapshot();
    });


    test('snapshot exception test', () => {
        const user =  {
            name: 'Pepito',
            id: Math.floor(Math.random()*30)
        }
        expect(user).toMatchSnapshot({
            id: expect.any(Number)
        })
    });

});

