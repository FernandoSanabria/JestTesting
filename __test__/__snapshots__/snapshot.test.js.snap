// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot fail snapthot test 1`] = `
Object {
  "createdAt": 2021-01-29T20:05:32.666Z,
  "id": 0,
}
`;

exports[`Snapshot faill snapthot test 1`] = `
Object {
  "id": 17,
}
`;

exports[`Snapshot rick object snapshot test 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Ricok Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Snapshot snapshot exception test 1`] = `
Object {
  "id": Any<Number>,
  "name": "Pepito",
}
`;
